nrcatalogtools/catalog.py

```diff
@@ -56,20 +79,43 @@ def get(self, sim_name):
                 f"Simulation {sim_name} not found in catalog."
                 f"Please check that it exists"
             )
-        filepath = self.waveform_filepath_from_simname(sim_name)
-        if not os.path.exists(filepath) or os.path.getsize(filepath) == 0:
-            if self._verbosity > 1:
-                print(
-                    f"..As data does not exist in cache:"
-                    f"  (in {filepath}),\n"
-                    f"..we will now download it from"
-                    " {}".format(self.waveform_url_from_simname(sim_name))
-                )
-            self.download_waveform_data(sim_name)
         metadata = self.get_metadata(sim_name)
         if type(metadata) is not dict and hasattr(metadata, "to_dict"):
             metadata = metadata.to_dict()
-        return waveform.WaveformModes.load_from_h5(filepath, metadata=metadata)
+
+        if quantity.lower() == "waveform":
+            filepath = self.waveform_filepath_from_simname(sim_name)
+            if not os.path.exists(filepath) or os.path.getsize(filepath) == 0:
+                if self._verbosity > 1:
+                    print(
+                        f"..As data does not exist in cache:"
+                        f"  (in {filepath}),\n"
+                        f"..we will now download it from"
+                        " {}".format(self.waveform_url_from_simname(sim_name))
+                    )
+                self.download_waveform_data(sim_name)
+            return waveform.WaveformModes.load_from_h5(filepath, metadata=metadata)
+        elif quantity.lower() == "psi4":
+            filepath = self.psi4_filepath_from_simname(sim_name)
+            if not os.path.exists(filepath) or os.path.getsize(filepath) == 0:
+                if self._verbosity > 1:
+                    print(
+                        f"..As data does not exist in cache:"
+                        f"  (in {filepath}),\n"
+                        f"..we will now download it from"
+                        " {}".format(self.psi4_url_from_simname(sim_name))
+                    )
+                self.download_psi4_data(sim_name)
+            try:
+                return waveform.WaveformModes.load_from_h5(filepath, metadata=metadata)
+            except:
```

Comment:
Please add the specific `Error` that needs to be caught.

Reply:
@adivijaykumar done. can you merge this please?